crates/ethereum/evm/src/execute.rs

```diff
@@ -841,4 +869,78 @@ mod tests {
             .unwrap();
         assert_eq!(parent_beacon_block_root_storage, U256::from(0x69));
     }
+
+    #[test]
+    fn eip_7002() {
+        let chain_spec = Arc::new(
+            ChainSpecBuilder::from(&*MAINNET)
+                .shanghai_activated()
+                .with_fork(Hardfork::Prague, ForkCondition::Timestamp(0))
+                .build(),
+        );
+
+        let mut db = create_state_provider_with_withdrawal_requests_contract();
+
+        let secp = Secp256k1::new();
+        let sender_key_pair = Keypair::new(&secp, &mut generators::rng());
+        let sender_address = public_key_to_address(sender_key_pair.public_key());
```

Comment:
why is this needed?

to generate a random address?

can we use Address::random?

Reply:
i'm signing the transaction with it